File: Bubble Sort/src/Bubble sort.c
void bubbleSort(int array[], int size) {
    for (int idx = 0; idx < size - 1; idx++) {
        int swapped = 0;
        for (int i = 0; i < size - idx - 1; i++) {
            if (array[i] > array[i + 1]) {
                int tmp = array[i];
                array[i] = array[i + 1];
                array[i + 1] = tmp;

                swapped = 1;
            }
        }
        if (swapped == 0) {
            break;
        }
    }
}

File: Bubble Sort/src/BubbleSort.java
public class BubbleSort{

    static void bubbleSort(int array[]) {
        int size = array.length;


        for (int i = 0; i < (size-1); i++) {

            boolean swapped = false;

            for (int j = 0; j < (size-i-1); j++) {

                if (array[j] > array[j + 1]) {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;

                    swapped = true;
                }
            }
            if (!swapped)
                break;
        }
    }

}
